### PR TITLE
Additional rmb command settings options

### PR DESCRIPTION
**Pull request by scrangos**
Adds settings for issuing target and attack as if it had alt pressed with rmb drag.  Also adds the ability to also issue line formation fight with the rmb drag.

### CODE REVIEW

**Comment by GoogleFrog** [comment]
I don't see why this is just for some commands. Shouldn't the setting simply make all RMB form line commands?

**Comment by GoogleFrog** [comment]
I don't see why a list of commands to apply the change to is required. I'd just do all of them. This widget usually doesn't care about what a command actually does or how it is issued. There should be a single consistent UI unless there is a good reason to do otherwise. One example of doing otherwise is the difference in handling between commands with an area mode and commands without an area mode. But in this case, since nothing else happens on RMB with a command selected, I don't see why all the commands wouldn't do the same thing.

The only difference between set target and set target circle is that one draws a circle area while the other draws a box. Neither of them is "the one that works". Some poweruser bound the non-circle set target because they wanted rectangle select rather than a circle. But as far as custom formations is concerned, they're just commands that can be issued on a line.

**Comment by scrangos** [comment]
Oh, I meant that in the option definition when I specify attack and set target I had tried the one that didnt say area first but when i tried to issue the command ingame it didnt work.  so the command bound to T ingame corresponds to the area circle im guessing since the other one didn't do anything.

Alright Ill change it to be a global implementation.

**Comment by scrangos** [comment]
Okay I updated the implementation

File: LuaUI/Widgets/cmd_customformations2.lua [hunk 1]
```diff
@@ -657,7 +671,10 @@ local function StopCommandAndRelinquishMouse()
 end
 
 function widget:MouseRelease(mx, my, mButton)
-	if (mButton == 1 or mButton == 3) and (not usingRMB) == (mButton == 3) then
+	--copied from press to check activecmd
+	local _, activeCmdID = spGetActiveCommand()
```
Comment (GoogleFrog) [inline LuaUI/Widgets/cmd_customformations2.lua:675]: This is redundant now right?

Comment (scrangos) [reply LuaUI/Widgets/cmd_customformations2.lua:675]: Is it, you're right. ill update the branch

**Comment by GoogleFrog** [comment]
Right clicking no longer cancels the command. It issues a point command.

**Comment by GoogleFrog** [comment]
It's weird that usingRMB is false when I use the right mouse button.

**Comment by scrangos** [comment]
Well, I loaded the game again with your fixed code, I tried all the combinations I could think of and it seems to be fixed.  Anything else that I can do?

**Comment by scrangos** [comment]
Hold up, theres something im not sure is wrong with plane context targetting behavior, need to test it without and with.

**Comment by scrangos** [comment]
Kay fixed it, had to add it to the commands that get overriden if the press and release are on the same entity.  Has the behavior that if you move the mouse off the unit and put it back before release it overrides, but it also does that with guard so I'm assuming that is the intended behavior.

File: LuaUI/Widgets/cmd_customformations2.lua [hunk 2]
```diff
@@ -131,6 +131,8 @@ local requiresAlt = {
 local overrideCmds = {
 	[CMD.GUARD] = CMD_RAW_MOVE,
 	[CMD_WAIT_AT_BEACON] = CMD_RAW_MOVE,
+	[CMD.ATTACK] = 20,
```
Comment (GoogleFrog) [inline LuaUI/Widgets/cmd_customformations2.lua:134]: Replace 20 with the command it is meant to be.

File: LuaUI/Widgets/cmd_customformations2.lua [hunk 3]
```diff
@@ -566,6 +569,10 @@ function widget:MousePress(mx, my, mButton)
 		
 		usingContextCommand = true
 		usingRMB = true
+		Spring.Echo("usingContextCommand")
```
Comment (GoogleFrog) [inline LuaUI/Widgets/cmd_customformations2.lua:572]: Echos

Comment (scrangos) [reply LuaUI/Widgets/cmd_customformations2.lua:572]: the heck, i thought i did find echo and i removed them all.
i tried using the attack line there but i got a not defined error, and CMD_RAW_MOVE isnt defined in that file.  wasnt sure what to do. still am not, should i just define it in the same file as a global?

**Pull request merged**